84 - improves synchronize high memory usage

jonjburgess opened the pull request:
Develop branch doesn't exist. So this is direct to master.

Added `stream.pause()` and `stream.resume()` to improve high memory usage.

Local tests

The app itself baseline ram usage is around ~80-100mb.
- Pre pull request = 2000 docs - ~360mb
- Post pull request = 2000 docs - ~140mb

There is still some data left in the ram but this solves a high percentage of it.

Also added a `batchSize` to mongo `stream` which saved ~10-20mb

Fix for #84

guumaster commented:
Good improvement. Although, there is an `options.bulk`, what if you use that instead of 50 hardcoded?

jonjburgess commented:
Good call. I have added `bulk.batch` to the bulk item, this defaults to 50. Also changed `bulk` to default each parameter individually incase one and not all parameters have been missed.
